workflows/scrnaseq.nf

```diff
@@ -67,13 +67,8 @@ workflow SCRNASEQ {
     ch_salmon_index = params.salmon_index ? file(params.salmon_index) : []
 
     //star params
-    if (params.star_index) {
-        ch_star_index = Channel.fromPath(params.star_index).map{f -> [[id:f.baseName], f] }
-    } else if (params.genome ) {
-        ch_star_index = Channel.fromPath(params.genomes[params.genome]["star"]).map{ f -> [[id:f.baseName], f] }
-    } else {
-        ch_star_index = []
-    }
+    star_index = params.star ? file(params.star_index, checkIfExists: true) : ( params.genome ? file( getGenomeAttribute('star'), checkIfExists: true ) : [] )
+    ch_star_index = [[id: star_index.baseName], star_index]
```

Comment:
```
params.star_index = getGenomeAttribute('star')
ch_star_index = [[id: star_index.baseName], file(params.star_index, checkIfExists: true)]
```

You are utilizing `params.star` which I have not seen elsewhere in the pipeline

Reply:
dammit, good catch

Reply:
I'll fix it in a separate PR

Reply:
Keep in mind:

> My proposed approach does not overwrite params.star_index with an igenomes index if it has already been set manually, because nextflow does not allow overwriting parameters.

Reply:
> params.star_index = getGenomeAttribute('star')

But that logic seems to overwrite `params.star_index` with the genome even when `star_index` is explicitly set. I find that a bit strange... or am I overlooking something here?

Reply:
ah, just saw your edit above now

Reply:
That simplifies things

Reply:
:100:

Reply:
Feel free to request my review on a new PR